DirectProgramming/DPC++FPGA/Tutorials/Tools/experimental/platform_designer/Readme.md

```diff
@@ -0,0 +1,274 @@
+
+   4. Make sure you choose an appropriate device. See **Board-specific Considerations** above.
+
+3. Copy the oneAPI-generated IP to the Intel Quartus® Prime project. This design uses host pipes, which generates additional internal SYCL kernels. The `fpga_export` build target uses the `-fsycl-device-code-split=per_kernel` flag to separate these additional kernels from your kernel, but these kernels will have their own reports and associated RTL. You will therefore need to hunt for the `.prj_X` directory that contains the IP you want to use in your design.
+
+   You can identify the correct `.prj_X` folder by looking for the one that contains a `*_di_inst.v` whose interfaces match your kernel. For example, in this project, `add_xample.fpga_ip.prj_1` is the correct `.prj_x` directory, because `add_example_fpga_ip_1_di_inst.v` contains only a CSR Agent interface in addition to the clock/reset signals:
+   
+   ```
+   add_example_fpga_ip_1_di add_example_fpga_ip_1_di_inst (
+     // Interface: clock (clock end)
+     .clock                          ( ), // 1-bit clk input
+     // Interface: clock2x (clock end)
+     .clock2x                        ( ), // 1-bit clk input
+     // Interface: resetn (conduit end)
+     .resetn                         ( ), // 1-bit data input
+     // Interface: device_exception_bus (conduit end)
+     .device_exception_bus           ( ), // 64-bit data output
+     // Interface: kernel_irqs (interrupt end)
+     .kernel_irqs                    ( ), // 1-bit irq output
+     // Interface: csr_ring_root_avs (avalon end)
+     .csr_ring_root_avs_read         ( ), // 1-bit read input
+     .csr_ring_root_avs_readdata     ( ), // 64-bit readdata output
+     .csr_ring_root_avs_readdatavalid( ), // 1-bit readdatavalid output
+     .csr_ring_root_avs_write        ( ), // 1-bit write input
+     .csr_ring_root_avs_writedata    ( ), // 64-bit writedata input
+     .csr_ring_root_avs_address      ( ), // 5-bit address input
+     .csr_ring_root_avs_byteenable   ( ), // 8-bit byteenable input
+     .csr_ring_root_avs_waitrequest  ( )  // 1-bit waitrequest output
+   );
+   ```
+
+   Linux:
+
+   ```
+   $> cd .. # navigate to project root if not there already
+   $> cp -r add-oneapi/build/add.fpga_ip_export.prj_1/ add-quartus/
+   ```
+
+   Windows:
+
+   ```
+   > cd .. # navigate to project root if not there already
+   > xcopy add-oneapi\build\add.fpga_ip_export.prj_1\ add-quartus\add.fpga_ip_export.prj_1 /e /s /i
+   ```
+
+4. Correct the generated `_hw.tcl` file by running the `*_di_hw_tcl_adjustment_script.py` script in the .prj directory.
+
+   Linux/Windows:
+
+   ```
+   $> cd add.fpga_ip_export.prj_1
+   $> python add_fpga_ip_export_1_di_hw_tcl_adjustment_script.py 
+   Success! Adjusted add_fpga_ip_export_1_di_hw.tcl file!
+   The adjustment log is in: adjustments_di_hw_tcl.log
+   The original file is in: add_fpga_ip_export_1_di_hw.tcl_original
+   ```
+
+5. Create the Platform Designer system.
```

Comment:
There's a lot of fiddly GUI stuff in here.  I know we want to teach users how to do this stuff, but maybe it would be a good idea to also include a finished version of all the relevant files they can use direcctly?

Reply:
I included that in an initial version, Yohann recommended I remove it and instead guide users on how to make the necessary quartus files.

Reply:
Totally agree the tutorial should guide users on how to do it, but wouldn't it be best to do both?

Reply:
One problem with including Quartus design files is that quartus design files get out of date, so unless we update them every release, quartus will whine about out-of-date IPs. This is a general problem for vertical IP Authoring designs that we will need to figure out in the future anyway, the solution will probably be to document that the quartus design files target a specific version of quartus, and you may need to modify them to work on a different version of quartus.

Reply:
OK, that makes sense.